### ISSUE TITLE
EAV module may generate deprecation error under PHP 8.1

### ISSUE THREAD

**Issue by speller** [magento/magento2#35856]
### Preconditions and environment

- Magento version 2.4.4
- PHP 8.1

### Steps to reproduce

Not really sure about the exact steps, but I'm checking for an attribute uniqueness before saving the customer user:
```
class Foo extends AbstractBackend
{
    protected function checkUniqueUsername(Customer $object)
    {
        $attribute = $this->getAttribute();
        $entity = $attribute->getEntity();
        if (!$entity->checkAttributeUniqueValue($attribute, $object)) {
            throw new AlreadyExistsException(__('Account with the same attribure already exist'));
        }
    }
}
```

### Expected result

Work fine under 2.4.3 + PHP 7.4

### Actual result

```none
Deprecated Functionality: trim(): Passing null to parameter #1 ($string) of type string is deprecated in /magento/vendor/magento/module-eav/Model/Entity/AbstractEntity.php on line 947
```

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by shikhamis11** [comment]
@magento give me 2.4-develop instance with edition ee

**Comment by engcom-Lima** [comment]
Hi @speller ,

Thanks for your contribution and collaboration.

Kindly provide us exact steps to reproduce the issue.
May be issue got fixed in the latest branch as this kind of error ` Deprecated Functionality: trim(): Passing null to parameter #1 ($string)` reported by many reporters. 
Please test in latest `2.4-develop` of Magento and let us know if you are facing any issue.

Thanks

**Comment by speller** [comment]
@engcom-Lima I didn't test in the dev version yet. But I think the issue appears when the customer entity has a custom attribute but is saved without setting this attribute value.

I'm using the following code to add this attribute:

```php
    public function install(ModuleDataSetupInterface $setup, ModuleContextInterface $context)
    {
        $this->logger->info("Installing Auth0 ID custom customer attribute.");
        $this->moduleDataSetup->getConnection()->startSetup();

        $this->customerSetup->addAttribute(
            Customer::ENTITY,
            Auth0Id::ATTR_NAME,
            [
                'type' => 'varchar',
                'label' => 'Auth0 ID',
                'input' => 'text',
                'backend' => Auth0Id::class,
                'required' => false,
                'visible' => false,
                'user_defined' => true,
                'unique' => true,
                'sort_order' => 150,
                'position' => 150,
                'system' => false,
            ]
        );

        $this->customerSetup->addAttributeToSet(
            CustomerMetadataInterface::ENTITY_TYPE_CUSTOMER,
            CustomerMetadataInterface::ATTRIBUTE_SET_ID_CUSTOMER,
            null,
            Auth0Id::ATTR_NAME,
        );

        $attribute =
            $this->customerSetup->getEavConfig()
                ->getAttribute(Customer::ENTITY, Auth0Id::ATTR_NAME)
                ->addData([
                    'used_in_forms' => [
                        'adminhtml_customer',
                    ],
                    'is_used_in_grid' => 1,
                    'is_visible_in_grid' => 1,
                    'is_filterable_in_grid' => 1,
                    'is_searchable_in_grid' => 1,
                ]);
        $this->attributeResource->save($attribute);
        $this->moduleDataSetup->getConnection()->endSetup();
        $this->logger->info("Attribute Auth0 ID successfully installed.");
    }
```

**Comment by engcom-Hotel** [comment]
Hello @speller,

Thanks for the report and collaboration!

We have tried to reproduce the issue in Magento's development branch i.e. 2-4-develop but for us the issue is not reproducible. We have made a custom module in order to reproduce the issue. The same is attached below:

[Magz.zip](https://github.com/magento/magento2/files/9415273/Magz.zip)

We have tried to reproduce the issue with PHP 8.1;

<img width="610" alt="image" src="https://user-images.githubusercontent.com/51681618/186397715-ae77e336-7699-426d-9d3a-e91b34c7b84a.png">

Please let us know in case we have missed anything.

Thanks

**Comment by speller** [comment]
@engcom-Hotel sorry for a late reply. I can confirm the issue is solved in 2.4.5. The line causing the warning was changed from 

```php
                'value' => trim($value),
```

to 

```php
                'value' => $value !== null ? trim($value) : '',
```

**Comment by engcom-Hotel** [comment]
Hello @speller,

Thanks for the conforming that issue has been resolved in the latest release version of Magento. Hence we are closing this issue.

Thanks
